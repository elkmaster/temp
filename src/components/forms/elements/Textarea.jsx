import React from 'react';
import PropTypes from 'prop-types';

import '../../../styles/components/forms.css';

const Textarea = ({
  input,
  meta: { touched, error },
  ...rest
}) => (
  <div>
    <textarea {...input} {...rest} />
    <div className="input-invalid">
      {touched && error}
    </div>
  </div>
);

Textarea.propTypes = {
  meta: PropTypes.object.isRequired,
  input: PropTypes.object.isRequired,
};

export default Textarea;
