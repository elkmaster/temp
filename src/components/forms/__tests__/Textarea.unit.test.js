import React from 'react';
import { shallow } from 'enzyme';
import Textarea from '../elements/Textarea';

const setup = (propsOverrides) => {
  const props = Object.assign({
      input: {},
      meta: {
        touched: false,
        error: '',
      },
    },
    propsOverrides);

  const wrapper = shallow(<Textarea {...props} />);

  return {
    props,
    wrapper,
    validationMessage: wrapper.find('.input-invalid'),
  };
};

describe('Textarea', () => {
  it('should render correctly', () => {
    const { wrapper } = setup();
    expect(wrapper).toMatchSnapshot();
  });

  describe('validation', () => {
    it('should display validation error when touched', () => {
      const { validationMessage } = setup({
        meta: {
          touched: true,
          error: 'Validation error',
        },
      });

      expect(validationMessage.text()).toBe('Validation error');
    });

    it('should not display validation error when untouched', () => {
      const { validationMessage } = setup({
        meta: {
          touched: false,
          error: 'Validation error',
        },
      });

      expect(validationMessage.text()).toBe('');
    });
  });
});
