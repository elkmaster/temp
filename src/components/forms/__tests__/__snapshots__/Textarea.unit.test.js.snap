// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Textarea should render correctly 1`] = `
<div>
  <textarea />
  <div
    className="input-invalid"
  />
</div>
`;
